pkg/controllers/status/cluster_status_controller.go

```diff
@@ -330,18 +331,19 @@ func getAPIEnablements(clusterClient *util.ClusterClient) ([]v1alpha1.APIEnablem
 	return apiEnablements, nil
 }
 
-func getNodeSummary(informerManager informermanager.SingleClusterInformerManager) (v1alpha1.NodeSummary, error) {
-	var nodeSummary = v1alpha1.NodeSummary{}
+func getClusterSummary(informerManager informermanager.SingleClusterInformerManager) (*v1alpha1.NodeSummary, *v1alpha1.ResourceSummary, error) {
```

Comment:
Why not split this function to `getNodeSummary` and `getResourceSummary`?

Reply:
If splitting, we should list all nodes twice.

Reply:
Sorry, I can't see why we list `nodes` twice. Did I miss something?

Reply:
Oh, I see it is `allocatable := getClusterAllocatable(nodes)`.

Reply:
But I still prefer to split this function, we can get `Nodes` and `Pods` first, then calculate `NodeSummary` and `ResourceSummary` respectively, which will make it easier to test by unit test.

Reply:
That's reasonable.